src/sonic-daemon-base/sonic_daemon_base/daemon_base.py

```diff
@@ -126,21 +129,32 @@ def signal_handler(self, sig, frame):
     # Returns platform and hwsku
     def get_platform_and_hwsku(self):
         try:
-            proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
-                                    stdout=subprocess.PIPE,
-                                    shell=False,
-                                    stderr=subprocess.STDOUT)
-            stdout = proc.communicate()[0]
-            proc.wait()
-            platform = stdout.rstrip('\n')
-
-            proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-d', '-v', HWSKU_KEY],
-                                    stdout=subprocess.PIPE,
-                                    shell=False,
-                                    stderr=subprocess.STDOUT)
-            stdout = proc.communicate()[0]
-            proc.wait()
-            hwsku = stdout.rstrip('\n')
+            platform = hwsku = None
+            if os.path.exists(SONIC_ENV_FILE):
+                with open(SONIC_ENV_FILE, "r") as env_file:
+                    for line in env_file:
+                        if PLATFORM_ENVVAR in line:
+                            platform = line.split('=')[1]
+                        if HWSKU_ENVVAR in line:
+                            hwsku = line.split('=')[1]
+
+            if not platform: 
+                proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
+                                        stdout=subprocess.PIPE,
+                                        shell=False,
+                                        stderr=subprocess.STDOUT)
+                stdout = proc.communicate()[0]
+                proc.wait()
```

Comment:
Your code is ok, but the original code has some questionable items.
1. communicate() will wait for process terminate. So second .wait() is not required.
2. I don't like a mix of stdout and stderr. I would split them.